osidb/models.py

```diff
@@ -2733,6 +2747,83 @@ def __str__(self):
         return self.username
 
 
+class FlawDraft(FlawMixin, AlertMixin):
```

Comment:
I am thinking about having `Flaw` and `FlawDraft` modeled in this way. If we had one (probably draft) as a parent of the other it would result in a lot of flaws with number of empty attributes and other disadvantages. This is maybe better internally. However externally as a data consumer I should be able to query for eg. all flaws of some source and obtain both draft and regular flaws and probably even rejected if no specified otherwise. Draft is should be considered as a first state of the flaw rather then a different entity.

Reply:
Going with a Polymorphic model was my first choice too but it complicates the "promotion" and "rejection" of drafts, the whole point of having a separate model is that you have the bare minimum amount of fields required in order to call it a proper Flaw, and promoting to an actual Flaw will trigger all of Flaw's validations.

Having it as a polymorphic model would imply, at the very least, a field to keep track of rejection/approval status in the polymorphic model itself and then a `save` override that would apply validations or not depending on the state.

I think in its current form it's sound from a modelling point of view, we can argue about the usefulness of displaying all draft flaws together with real flaws (there is a lot, and I mean a lot of vulnerabilities that will never materialize because they don't affect Red Hat products) but I don't see how including those together with actual flaws would be relevant.

Reply:
So I will leave it like this for now and think about this more in depth, but I think the current approach is correct:

* There are currently 97k assembler tickets and ~40k OSIDB flaws, merging them together into a single endpoint will not only make it a mess for users to navigate through, but will also impact our already poor performance.
* OSIDB's main mission is to display flaws that impact Red Hat products, this can be true for IR-created `FlawDraft`, but it is not the case for the vast majority of assembler tickets.
* Using a polymorphic model complicates things at the API level, because `FlawDraft` and `Flaw` will have different schemas (a draft can include very little information), we'll have to cater to lowest common denominator (`FlawDraft`) which would make the `/flaws` endpoint backwards incompatible.

Since we're only working at the model level right now and not the API level, we can see how it goes after a first collector implementation at the very least, and then we can always change the model before implementing the API.

Reply:
OK. Let us see. I still think that an option of querying all the flaws at once is going to be important but we can figure it out later. It should be no regression with regards to SFM2 where the ASM and Flaws are even further from each other. On the other hand the proper merge seems to me as on opportunity how to simplify this from the user point of view and the task should be to make the tools can handle it. I am not going to block this PR on this.